bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -129,6 +135,103 @@ enum ClusterConfig {
     },
 }
 
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Config {
```

Comment:
Can we get doc strings for all these structs? I think this is specifically an `eksctl` configuration struct, but a docstring would help. I think `EksctlConfig` might be a better name as well to signal we don't own this format, its intended to be used specifically as input to `eksctl`.